packages/data/index.d.ts

```diff
@@ -57,15 +60,18 @@ export function hasMany(): PropertyDecorator;
  * import { hasMany } from 'ember-decorators/data';
  *
  * export default Model.extend({
- *   @hasMany users: null
+ *   @hasMany('user') users: null,
+ *   @hasMany('car', { async: false }) cars: null
  * });
  * ```
  *
  * @function
  * @param {String} [type] - Type of the `hasMany` relationship
+ * @param {Object} [options] - Optional relationship options
  */
 export function hasMany<K extends keyof ModelRegistry>(
-  type: K
+  type: K,
```

Comment:
This is why I got the `attr` one wrong originally! I started by building out `hasMany` and `belongsTo` and didn’t flip the mental switch when I switched to `attr`.

Reply:
That's what I guessed had happened. It'd probably be better to call the first argument something like `transform` for `attr` and `modelName` for `hasMany`/`belongsTo`, but I left them as `type` for now to be consistent with the doc blocks in https://github.com/ember-decorators/ember-decorators/blob/master/packages/data/addon/index.js

`@types/ember-data` is inconsistent for [hasMany](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember-data/index.d.ts#L82) and [belongsTo](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember-data/index.d.ts#L70).

Reply:
Also, I intentionally omitted the Sync/Async distinction that `@types/ember-data` has because we can't currently affect the attribute's type with decorators anyways.